protos/flyteidl/service/dataproxy.proto

```diff
@@ -45,6 +45,12 @@ message CreateUploadLocationRequest {
   // generated path.
   // +required
   bytes content_md5 = 5;
+
+  // If present, data proxy will use this string in lieu of the md5 hash in the path. When the filename is also included
+  // this makes the upload location deterministic. The native url will still be prefixed by the upload location prefix
+  // in data proxy config. This option is useful when uploading multiple files.
+  // +optional
+  string filename_root = 6;
```

Comment:
This is getting confusing... Can we add docs on top of the message explaining all the different ways a name will be generated?

Reply:
done